### ISSUE TITLE
`result_large_err` doesn't trigger for async fn that returns a Result

### ISSUE THREAD

**Issue by beicause** [rust-lang/rust-clippy#17070]
### Summary

`result_large_err` doesn't trigger for async fn that returns a Result

### Lint Name

result_large_err

### Reproducer

I tried this code:

```rust
pub enum MyError {
    A([u8; 256]),
}

pub async fn test() -> Result<(), MyError> {
    Ok(())
}
```

I expected to see this happen like the non-async fn:
```
warning: the `Err`-variant returned from this function is very large
  --> wgpu/src/backend/mod.rs:20:18
   |
16 |     A([u8;256]),
   |     ----------- the largest variant contains at least 256 bytes
...
20 | pub fn test() -> Result<(),MyError> {
   |                  ^^^^^^^^^^^^^^^^^^
   |
   = help: try reducing the size of `backend::MyError`, for example by boxing large elements or replacing it with `Box<backend::MyError>`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#result_large_err
   = note: `#[warn(clippy::result_large_err)]` on by default
```

Instead, No warnings happened.

### Version

```text
rustc 1.93.1 (01f6ddf75 2026-02-11)
binary: rustc
commit-hash: 01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf
commit-date: 2026-02-11
host: x86_64-unknown-linux-gnu
release: 1.93.1
LLVM version: 21.1.8
```
